redshells/model/graph_convolutional_matrix_completion.py

```diff
@@ -27,163 +28,132 @@ def _convert_sparse_matrix_to_sparse_tensor(x):
-            self.item_adjustment = [tf.div_no_nan(1., tf.sparse.reduce_sum(m, axis=0)) for m in self.input_adjacency_matrix]
-            self.rating = tf.constant(rating.reshape((-1, 1)), dtype=np.float32)
-            # features
-            self.input_user_features = [
-                tf.placeholder(dtype=np.float32, shape=[None, size], name=f'user_features_{i}') for i, size in enumerate(user_feature_sizes)
-            ]
-            self.input_item_features = [
-                tf.placeholder(dtype=np.float32, shape=[None, size], name=f'item_features_{i}') for i, size in enumerate(item_feature_sizes)
-            ]
-            # adjusted adjacency matrix
-            self._adjust_adjacency_matrix(normalization_type)
-
-            # C X
-            # (n_user, item_feature_size)
-            self.item_cx = [tf.sparse.slice(m, [0, 0], [n_user, n_item]) for m in self.adjusted_adjacency_matrix]
-            # (n_item, user_feature_size)
-            self.user_cx = self.adjusted_adjacency_matrix_transpose
-
-            # encoder
-            self.common_encoder_layer = self._simple_layer(encoder_size)
-
-            self.item_encoder_hidden = self._encoder(
-                feature_size=n_user,
-                encoder_hidden_size=encoder_hidden_size,
-                n_rating=n_rating,
-                cx=self.user_cx,
-                dropout=self.input_dropout,
-                weight_sharing=weight_sharing,
-                edge_size=self.input_edge_size,
-                prefix='item')
-
-            self.user_encoder_hidden = self._encoder(
-                feature_size=n_item,
-                encoder_hidden_size=encoder_hidden_size,
-                n_rating=n_rating,
-                cx=self.item_cx,
-                dropout=self.input_dropout,
-                weight_sharing=weight_sharing,
-                edge_size=self.input_edge_size,
-                prefix='user')
-
-            self.item_encoder = self.common_encoder_layer(self.item_encoder_hidden)
-            self.user_encoder = self.common_encoder_layer(self.user_encoder_hidden)
-            self.user_encoder = tf.gather(self.user_encoder, self.input_user)
-            self.item_encoder = tf.gather(self.item_encoder, self.input_item)
-
-            self._add_user_feature(encoder_hidden_size, encoder_size)
-            self._add_item_feature(encoder_hidden_size, encoder_size, ignore_item_embedding)
-
-            # decoder
-            self.output = self._decoder(encoder_size=encoder_size, n_rating=n_rating, user_encoder=self.user_encoder, item_encoder=self.item_encoder)
-
-            # output
-            self.probability = tf.nn.softmax(self.output)
-            self.expectation = tf.matmul(self.probability, tf.reshape(self.rating, shape=(-1, 1)))
-            self.rmse = tf.sqrt(tf.reduce_mean(tf.math.square(self.expectation - tf.reshape(tf.to_float(self.input_rating), shape=(-1, 1)))))
+class GraphABC(ABC):
```

Comment:
@mski-iksm 
In my opinion, `AbstractGraph` sounds better than `GraphABC` in terms of the name of the class because abbreviations should be avoided if possible.

Reply:
I've changed the class name to AbstractGraph.